src/util/taproot.rs

```diff
@@ -390,7 +392,7 @@ pub struct TaprootBuilder {
     // to a depth-first traversal of a tree). branch is only empty if no entries
     // have ever be processed. branch having length 1 corresponds to being done.
     //
-    branch: Vec<Option<NodeInfo>>,
+    pub(crate) branch: Vec<Option<NodeInfo>>,
```

Comment:
I don't think it's great to have this `pub(crate)` if there's an invariant. Can we restrict the scope somehow?

Reply:
Will add a pub(crate) accessor to the field as I don't really need to write to that field.